webapp/api/requests.py

```diff
@@ -1,15 +1,18 @@
-# Core packages
 import os
 from urllib.parse import urlparse
 
-# Third-party packages
 import requests
-import requests_cache
-import prometheus_client
 
-# Local packages
-from webapp.api.exceptions import ApiTimeoutError, ApiConnectionError
+import prometheus_client
+import requests_cache
+from pybreaker import CircuitBreaker, CircuitBreakerError
+from webapp.api.exceptions import (
+    ApiCircuitBreaker,
+    ApiConnectionError,
+    ApiTimeoutError,
+)
 
+db_breaker = CircuitBreaker(fail_max=100)
```

Comment:
I think a failure threshold of `100` is insanely large. Too large to be of any practical use. Unless we have some indication that 5 or more consecutive failure are actually incredibly common? If so that would be very concerning.

I'd also suggest that for this circuit breaker to be useful, its failure threshold should be less than the `failureThreshold` of the Kubernetes `readinessProbe`. Otherwise pods will be brought out of rotation before the circuit breaker kicks in. We don't seem to have set that setting in [our configs](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/), which means it must use the default `failureThreshold` of 3.

So maybe we should set this threshold to the default of 5, and then increase the Kubernetes threshold to a larger number? Or should we go the other way and set the circuit breaker to be super aggressive at a threshold of 2?

@jpmartinspt do you have any thoughts?

Reply:
Good point, your right that simple connection failures shouldn't effect k8s.

But timeouts still might. The timeout in the app requests is what now? 2 seconds? 3? The timeout for the readinessprobe is 5. We could easily hit that if a route makes multiple API calls, or if the timeout isn't working properly for some reason, or if there are delays elsewhere in the app.

So I still think it's true that any failure solutions in the app itself should have lower thresholds than those at the Kubernetes layer, for robustness.

Reply:
But 5 is fine, but I think we should up the k8s one.